src/backend/distributed/executor/local_executor.c

```diff
@@ -585,6 +588,39 @@ ExecuteLocalTaskPlan(PlannedStmt *taskPlan, char *queryString,
 }
 
 
+/*
+ * RecordNonDistTableAccessesForTask records relation accesses for the non-distributed
+ * relations that given task will access.
+ */
+static void
+RecordNonDistTableAccessesForTask(Task *task)
+{
+	ShardPlacement *taskPlacement = NULL;
+	foreach_ptr(taskPlacement, task->taskPlacementList)
+	{
```

Comment:
We do this call for each placement, but is the outcome ever different across placements of a reference table? Wouldn't we keep finding the same set of relations for every placement?

Reply:
Even more, as we are in the local executor, can it ever be possible that we have more than one task placements if we are dealing with a reference table ?
I think the code should look like that: 022efba